src/test/java/com/conveyal/gtfs/loader/JDBCTableWriterTest.java

```diff
@@ -421,7 +421,15 @@ public void shouldChangeShapeIdOnPatternUpdate() throws IOException, SQLExceptio
         // The output should contain a new backend-generated shape_id.
         PatternDTO updatedSharedPattern = mapper.readValue(sharedPatternOutput, PatternDTO.class);
         LOG.info("Updated pattern output: {}", sharedPatternOutput);
-        assertThat(updatedSharedPattern.shape_id, not(equalTo(sharedShapeId)));
+        String newShapeId = updatedSharedPattern.shape_id;
+        assertThat(newShapeId, not(equalTo(sharedShapeId)));
+        // Ensure that pattern record in database reflects updated shape ID.
+        assertThatSqlQueryYieldsRowCount(String.format(
+            "select * from %s.%s where shape_id='%s'",
```

Comment:
I think we're almost there, but this assertion would pass if somehow the pattern with `patternId` was somehow updated to have the value of `newShapeId`.  This could be improved by adding a where clause for the pattern id so that a more exact expectation of data integrity can happen.

Reply:
Are you suggesting that this where clause be changed to `where shape_id=$newShapeId AND pattern_id=$secondPatternId`?

Reply:
Yes, exactly.

Reply:
Great to see very specific assertions happening.